runtime/server/components.go

```diff
@@ -101,3 +145,37 @@ func (s *Server) getComponent(ctx context.Context, instanceID, name string) (*ru
 
 	return spec, nil
 }
+
+func (s *Server) getParsedComponent(ctx context.Context, instanceID, name string, args map[string]any) *string {
+	ctrl, err := s.runtime.Controller(ctx, instanceID)
+	if err != nil {
+		return nil
+	}
+	fmt.Println(name, args)
+
+	res, err := ctrl.Get(ctx, &runtimev1.ResourceName{Kind: runtime.ResourceKindComponent, Name: name}, false)
+	if err != nil {
+		return nil
+	}
+
+	blob, _, err := s.runtime.GetFile(ctx, instanceID, res.Meta.GetFilePaths()[0])
+	if err != nil {
+		return nil
+	}
```

Comment:
It's not safe to rely on file loading from APIs that are used for dashboard serving because the files may be unavailable (for example, if the Github clone takes a long time or Github is unavailable). Viewer roles also are not allowed to have any way of loading files for security reasons.

So it should only use fields in `res` (which are available for querying as soon as the server is up).